### ISSUE TITLE
Can RDPWrapper update be scheduled?

### ISSUE THREAD

**Issue by bellocarico** [stascorp/rdpwrap#539]
Since Windows is messing about with RDP more and more with all those updates, could and RDPWrapper check/update be installed in the scheduled task (like AutoKSD does per se) to make sure RDP is always enabled and listening?

Thanks :-)

**Comment by binarymaster** [comment]
Yes, you can create scheduled task that will run RDPWInst for example (with needed arguments).

**Comment by bellocarico** [comment]
Interesting.... I will give it a go!
Beside this, would you consider adding as "Add scheduled task" button/function from the GUI to do this automatically?
I think it would be a great idea to have the RDPWrapper installation resilient!

**Comment by mariagil1975** [comment]
Totally agree. That in the future this option is added in the installer.

**Comment by bellocarico** [comment]
Can this be added to the roadmap for the next release please?

**Comment by binarymaster** [comment]
@bellocarico I'm not planning to implement that for now. The project is open source, and anybody interested is welcome to implement it (Pull Requests are appreciated).

**Comment by bellocarico** [comment]
Ok, perhaps I can look into this myself and share the result. 
The RDPWinst is pretty straight forward but I can't find an option though to enable RDP as this is usually what it brakes after a windows update. 

See pic: [https://imgur.com/a/gJevGUG](url)

So how can I enable the RDP listening from command line?

Thanks

**Comment by bellocarico** [comment]
Ok so I made some progression. I have discovered that installing (from scratch) RDPWrapper does leave the system with the option **Enable Remote Desktop** set.

So I went creating a Scheduled Task. 

- Open the Task Scheduler utility
- On the right hand side select Create Task (don't use the Basic Task for this)

General:
- select your Admin username
- set: Run whether user is connected or not
- set: Run with highest privileges
- set Compatibility Windows10 (in my case)

Triggers:
- Daily at 5am (in my case)

Actions:
- **C:\Users\Administrator\Desktop\RDPWrapper\RDPWInst.exe** with argument **-u -k**
then add a second action on the very same tab:
- **C:\Users\Administrator\Desktop\RDPWrapper\RDPWInst.exe** with argument **-i -o**

Click Ok, provide the Admin password when requested and you're good to go.

A couple of notes:
- if you run the task manually the first time (you should) you can enable "All Task History"first  on the right hand side of the Task Scheduler. This will populate the history tab of the task. After you can see it works I suggest you disable the history again.
- if you run the task manually the state will go into "Running" and will stay there long time, you might need to right click and select refresh to verify that it actually goes back to "Ready"
- I have tried to export the task but for some odd reason the other Windows10 machined will not accept the .xml so had to create manually everywhere unfortunately...

**Comment by jaxjexjox** [comment]
> Since Windows is messing about with RDP more and more with all those updates, could and RDPWrapper check/update be installed in the scheduled task (like AutoKSD does per se) to make sure RDP is always enabled and listening?
> 
> Thanks :-)

Based on the frequency of Microsoft screwing this up, I wouls strongly consider adjusting yourself to the slower Windows 10 update channel to avoid it being damaged.

Hopefully, ideally - if you get your Windows 10 update several weeks (months?) after others, you may end up in a position where updating RDP wrap will end up working each time.  
Whereas now, if you update, you often find the Application is not 'ready' for the new release unfortunately.
